crowdsourcing/serializers/user.py

```diff
@@ -135,15 +137,18 @@ def create(self, **kwargs):
         user_profile = models.UserProfile()
         user_profile.user = user
         user_profile.save()
-        worker = models.Worker()
-        worker.profile = user_profile
-        worker.alias = username
-        worker.save()
-
-        requester = models.Requester()
-        requester.profile = user_profile
-        requester.alias = username
-        requester.save()
+
+        if 'is_worker' not in self.validated_data or self.validated_data['is_worker']:
```

Comment:
not quite. basically if these fields arent specified (i.e. via normal user creation) then 'is_worker' not in self.validated_data is True and  the Worker should be created by default. Otherwise self.validated_data['is_worker'] will exist and the Worker object will be created appropriately. Kind of a short circuit dealio.